services/scan/src/precinct_scanner_state_machine.ts

```diff
@@ -0,0 +1,854 @@
+      const scannerState = (() => {
+        switch (true) {
+          case state.matches('waiting_for_configuration'):
+            return 'unconfigured';
+          case state.matches('connecting'):
+            return 'connecting';
+          case state.matches('checking_initial_paper_status'):
+            return 'connecting';
+          case state.matches('error_disconnected'):
+            return 'disconnected';
+          case state.matches('reconnecting'):
+            return 'disconnected';
+          case state.matches('no_paper'):
+            return 'no_paper';
+          case state.matches('ready_to_scan'):
+            return 'ready_to_scan';
+          case state.matches('scanning'):
+            return 'scanning';
+          case state.matches('error_scanning'):
+            return 'scanning';
+          case state.matches('checking_scanning_completed'):
+            return 'scanning';
+          case state.matches('interpreting'):
+            return 'scanning';
+          case state.matches('ready_to_accept'):
+            return 'ready_to_accept';
+          case state.matches('accepting'):
+            return 'accepting';
+          case state.matches('error_accepting'):
+            return 'accepting';
+          case state.matches('accepted'):
+            return 'accepted';
+          case state.matches('needs_review'):
+            return 'needs_review';
+          case state.matches('returning'):
+            return 'returning';
+          case state.matches('checking_returning_completed'):
+            return 'returning';
+          case state.matches('returned'):
+            return 'returned';
+          case state.matches('rejecting'):
+            return 'rejecting';
+          case state.matches('checking_rejecting_completed'):
+            return 'rejecting';
+          case state.matches('rejected'):
+            return 'rejected';
+          case state.matches('calibrating'):
+            return 'calibrating';
+          case state.matches('calibrated'):
+            return 'calibrated';
+          case state.matches('error_jammed'):
+            return 'jammed';
+          case state.matches('error_both_sides_have_paper'):
+            return 'both_sides_have_paper';
+          case state.matches('error'):
+            return 'error';
+          default:
+            throw new Error(`Unexpected state: ${state.value}`);
+        }
+      })();
+      const { error, interpretation } = state.context;
+      // TODO log any errors, especially unexpected paper status/event or other unexpected errors
+      return {
+        state: scannerState,
+        interpretation,
+        error:
+          error instanceof PrecinctScannerError ? error.type : 'plustek_error',
+      };
+    },
+
+    scan: () => {
+      if (!machineService.state.matches('ready_to_scan')) {
+        return err(
+          new Error('Scanner must be in ready_to_scan state in order to scan')
+        );
+      }
```

Comment:
is this repetitive? Shouldn't the state machine be able to say "hey, I don't know how to scan right now." ?

Reply:
Yes, though the error isn't the most clear. I put this in to help abstract away the state machine. But maybe I'll try removing it to see what happens.

Reply:
Oh now I remember why I put this in. `send` won't fail when this happens, so the API request will always be successful. I think maybe that's ok, since there will be an error status within the state machine, which is probably better anyway since we handle that smartly on the frontend (vs failed API requests, which I don't think we handle well).